x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsResultProcessor.java

```diff
@@ -52,12 +52,25 @@ public void awaitForCompletion() {
     }
 
     public void process(AnalyticsProcess<AnalyticsResult> process) {
+        long totalRows = process.getConfig().rows();
+        LOGGER.info("Total rows = {}", totalRows);
+        long processedRows = 0;
+
         // TODO When java 9 features can be used, we will not need the local variable here
         try (DataFrameRowsJoiner resultsJoiner = dataFrameRowsJoiner) {
             Iterator<AnalyticsResult> iterator = process.readAnalyticsResults();
             while (iterator.hasNext()) {
                 AnalyticsResult result = iterator.next();
                 processResult(result, resultsJoiner);
+                if (result.getRowResults() != null) {
+                    processedRows++;
+                    progressTracker.writingResultsPercent.set(processedRows == totalRows ? 100 : (int) (processedRows * 100.0 / totalRows));
```

Comment:
```suggestion
                    progressTracker.writingResultsPercent.set(processedRows >= totalRows ? 100 : (int) (processedRows * 100.0 / totalRows));
```

Similar reason to before, just a little insurance.